cvelib/cve_api.py

```diff
@@ -70,19 +71,37 @@ def post(self, path: str, **kwargs) -> requests.Response:
     def put(self, path: str, **kwargs) -> requests.Response:
         return self.http_request("put", path, **kwargs)
 
-    def publish(self, cve_id: str, cve_json: dict):
+    def publish(self, cve_id: str, cve_json: dict) -> Tuple[bool, dict]:
         """Publish a CVE from a JSON object representing the CNA container data."""
         cve_json = {"cnaContainer": cve_json}
-        response = self.post(f"cve/{cve_id}/cna", json=cve_json)
-        response.raise_for_status()
-        return response.json()
-
-    def reject(self, cve_id: str, cve_json: dict):
+        try:
+            response = self.post(f"cve/{cve_id}/cna", json=cve_json)
+            response.raise_for_status()
+            created = True
+        except requests.exceptions.HTTPError as e:
+            error = e.response.json()["error"]
+            if e.response.status_code != 403 or error != self.RECORD_EXISTS:
+                raise e
+            response = self.put(f"cve/{cve_id}/cna", json=cve_json)
```

Comment:
I'm on the fence whether this is a good idea. I can imagine as a developer using this library in my application I may want to call one or the other method (create vs update) explicitly instead of this function deciding the behavior for me. The CLI could definitely do that since it's user-facing and "we know better than users"™ :-) Thoughts?

To be more specific, I imagine in the CveApi interface we could have all four:

```
def publish_cve_record(): ...

def update_published_cve_record(): ...

def reject_cve_record(): ...

def update_rejected_cve_record(): ...
```

and in the CLI, you could abstract a single `publish` command to:

```
try:
    publish_cve_record()
except:
    update_published_cve_record()
```

Reply:
+1 from me, it's probably better to have cleaner separation between CLI logic (handle invalid states, pretty print errors) and API logic (return raw errors and don't try to be clever). Will fixi in a bit.